### DISCUSSION TITLE
lazy loading

### DISCUSSION THREAD

**Discussion by leostereo** [quasarframework/quasar#9842] [General - Components / Directives / etc] [answered]
Hi guys , I need to create some routes with a lot of static content.
Those routes are not allways being visited by users so I decided to try lazy loading (first time for me)
Reading from quasar documentation I understand I need use dinamyc imports on the routes.js (wich is already done) , and i think I need to do same with components within this route.
So I tryed as document with:
```
<script>
import { defineAsyncComponent } from 'vue'

//import basicCard from "../../components/activities/basicCard.vue";
import statementCard from "../../components/activities/statementsCard.vue";
import statementResultCard from "../../components/activities/statementsResultCard.vue";
import wellnessChartCard from "../../components/activities/wellnessChartCard.vue";
import DAY2 from "../../constants/activities/day2";

export default {
  name: "activity2",
  components: {
    basicCard: defineAsyncComponent(() => import('../../components/activities/basicCard.vue')),
    
    statementCard,
    statementResultCard,
    wellnessChartCard
  },

```
but im getting this:
`"export 'defineAsyncComponent' was not found in 'vue'`
Where should I declare this export? did not find related information.
Other:
Should I do same with external files /  constants as well ? how ? 
Can I confirm lazy loading is working on the browser console ? 

Thanks guys.
Leo.

**Comment by MilosPaunovic** [comment, marked as answer]
Hello @leostereo,

Check out this [issue](https://github.com/quasarframework/quasar/issues/9796) and this [particular](https://github.com/quasarframework/quasar/issues/9796#issuecomment-868208660) reply and see if it helps.

Just to be sure, you are using Quasar 2?

**Reply by leostereo** [reply]
Dear @MilosPaunovic , Im checking about version this is what I have: (quasar info out)
Important local packages:

```
  quasar - 1.15.19-beta.1 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app - 2.2.6 -- Quasar Framework local CLI
  @quasar/extras - 1.10.5 -- Quasar Framework fonts, icons and animations
  vue - 2.6.12 -- Reactive, component-oriented view layer for modern web interfaces.
  vue-router - 3.2.0 -- Official router for Vue.js 2
  vuex - 3.6.0 -- state management for Vue.js
```

im not sure wich is the real version 1.15.19 or 2.2.6.
should I upgrade ? 

Also tryed:
`import { defineComponent, defineAsyncComponent } from 'vue'`

but didn work.
Thanks for your help.
leandro

**Reply by MilosPaunovic** [reply]
Okay, so, you are using Quasar `v1`, but I suggest for you upgrade to the latest stable release of that version, which is `1.15.21`.

My best guess is that you confused [official docs](https://quasar.dev/), which now point to Quasar `v2` and use Vue 3 way of lazy loading components.

What you should check out is[ version one docs](https://v1.quasar.dev/), more specifically [this section](https://v1.quasar.dev/quasar-cli/lazy-loading#lazy-load-components) about lazy loading components in the Vue 2 way.

**Reply by leostereo** [reply]
Dear Milos , already updated , now it looks like:
Important local packages
  quasar - 1.15.21 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app - 2.2.10 -- Quasar Framework local CLI
  @quasar/extras - 1.10.8 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 2.6.14 -- Reactive, component-oriented view layer for modern web interfaces.
  vue-router - 3.5.1 -- Official router for Vue.js 2
  vuex - 3.6.2 -- state management for Vue.js
  
Documentation you pointed worked for me.
I disable cache on network tab on my dev-tools and can see now Im getting some 1,2,3,4,5 ...js separated files.
Does it mean it is working ? 
Thankyou very much.

**Reply by MilosPaunovic** [reply]
You can open network tab and check if those chunks are being loaded on the page where you've lazy loaded component in question. And yes, that would mean it works :smile:

**Reply by leostereo** [reply]
thanks thanks ...thanks
